distance.sql

```diff
@@ -1,28 +1,74 @@
 -- nullsec to low
-SELECT a.solarSystemID as sourceSystemID, a.solarSystemName as sourceSystem, a.security,  b.solarSystemID as destSystemID, b.solarSystemName as destSystem, b.security, MIN((a.x - b.x)*(a.x - b.x) + (a.y - b.y)*(a.y - b.y) + (a.z - b.z)*(a.z - b.z)) as distance
+SELECT a.solarSystemID AS sourceSystemID,
+       a.solarSystemName AS sourceSystem,
+       a.security,
+       ra.regionName as sourceRegion,
+       b.solarSystemID AS destSystemID,
+       b.solarSystemName AS destSystem,
+       b.security,
+       rb.regionName as destRegion,
+       MIN((a.x - b.x)*(a.x - b.x) + (a.y - b.y)*(a.y - b.y) + (a.z - b.z)*(a.z - b.z)) AS distance
 FROM mapSolarSystems AS a
 JOIN mapSolarSystems AS b
-WHERE b.security >= 0.0 AND ROUND(b.security, 1) < 0.5 AND a.security < 0.0  AND a.securityClass IS NOT NULL
--- Make sure destination system has a station
-AND (SELECT 1 FROM staStations as stations WHERE b.solarSystemID == stations.solarSystemID)
+LEFT JOIN mapRegions AS ra ON ra.regionId = a.regionId
+LEFT JOIN mapRegions AS rb ON rb.regionId = b.regionId
+WHERE b.security >= 0.0
+  AND ROUND(b.security, 1) < 0.5
+  AND a.security < 0.0
+  AND a.securityClass IS NOT NULL
+  AND a.regionId != 10000070  -- no asset safety for Pochven origins
+  AND  -- Make sure destination system has a station
+    (SELECT 1
+     FROM staStations AS stations
+     WHERE b.solarSystemID == stations.solarSystemID)
 GROUP BY a.solarSystemID
 
 UNION
+
 -- lowsec to low
-SELECT a.solarSystemID as sourceSystemID, a.solarSystemName as sourceSystem, a.security,  b.solarSystemID as destSystemID, b.solarSystemName as destSystem, b.security, MIN((a.x - b.x)*(a.x - b.x) + (a.y - b.y)*(a.y - b.y) + (a.z - b.z)*(a.z - b.z)) as distance
+SELECT a.solarSystemID AS sourceSystemID,
+       a.solarSystemName AS sourceSystem,
+       a.security,
+       ra.regionName as sourceRegion,
+       b.solarSystemID AS destSystemID,
+       b.solarSystemName AS destSystem,
+       b.security,
+       rb.regionName as destRegion,
+       MIN((a.x - b.x)*(a.x - b.x) + (a.y - b.y)*(a.y - b.y) + (a.z - b.z)*(a.z - b.z)) AS distance
 FROM mapSolarSystems AS a
 JOIN mapSolarSystems AS b
-WHERE b.security >= 0.0 AND ROUND(b.security, 1) < 0.5 AND a.security >= 0.0 AND ROUND(a.security, 1) < 0.5
--- Make sure destination system has a station
-AND (SELECT 1 FROM staStations as stations WHERE b.solarSystemID == stations.solarSystemID)
+LEFT JOIN mapRegions AS ra ON ra.regionId = a.regionId
+LEFT JOIN mapRegions AS rb ON rb.regionId = b.regionId
```

Comment:
Is there a reason to care about the regions?

Reply:
I'm needlejacking in Catch right now, and want to asset safety stuff back to k-space at some point but don't care about exactly where in the origin region I do it from -- so wanted to see what my destination choices were, given that origin.

Reply:
I think there's other use cases too -- "as Fortress Delve falls, where will the stuff mostly wind up?"